elitzur-core/src/main/scala/com/spotify/elitzur/validators/Validator.scala

```diff
@@ -261,66 +268,57 @@ object Validator extends Serializable {
                      config: ValidationRecordConfig = DefaultRecordConfig)
                     (implicit reporter: MetricsReporter): PostValidation[T] = {
     val cs = new Array[Any](validatorAccessors.length)
-    var i = 0
     var atLeastOneValid = false
     var atLeastOneInvalid = false
 
-    while (i < validatorAccessors.length) {
-      val accessor = validatorAccessors(i)
-      val v =
-        if (!accessor.validator.isInstanceOf[IgnoreValidator[_]]) {
-          val name = new JStringBuilder(path.length + accessor.label.length)
-            .append(path).append(accessor.label).toString
-          val o = if (accessor.validator.isInstanceOf[FieldValidator[_]]) {
-            val v = accessor.validator.validateRecord(Unvalidated(accessor.value))
-            val validationType = accessor.validator.asInstanceOf[FieldValidator[_]].validationType
-
-            val c = config.fieldConfig(name)
-            if (v.isValid) {
-              if (c != NoCounter) {
-                reporter.reportValid(
-                  outermostClassName,
-                  name,
-                  validationType
-                )
-              }
-            } else if (v.isInvalid) {
-              if (c == ThrowException) {
-                throw new DataInvalidException(
-                  s"Invalid value ${v.forceGet.toString} found for field $path${accessor.label}")
-              }
-              if (c != NoCounter) {
-                reporter.reportInvalid(
-                  outermostClassName,
-                  name,
-                  validationType
-                )
-              }
-            }
-            v
-          } else {
-            accessor.validator.validateRecord(
-              Unvalidated(accessor.value),
-              new JStringBuilder(path.length + accessor.label.length + 1)
-                .append(path).append(accessor.label).append(".").toString,
-              Some(outermostClassName),
-              config
-            )
-          }
+    validatorAccessors.zipWithIndex.foreach {
```

Comment:
We explicitly want to avoid using the nicer scala functions here because there is a big performance hit. So this code should be mutable and use some barebones java like functionality since it is also used in the magnolia code (i.e. while loops instead of fors/maps, vars, mutable lists, if/else instead of case matching, etc)

Reply:
This is especially important since this core loop is being used directly in user code in the case of the derived validators. Meaning performance is a major concern

Reply:
Gotcha makes sense.